docs/design/autoware-interfaces/ad-api/types/autoware_ad_api_msgs/msg/steering_factor.md

```diff
@@ -0,0 +1,39 @@
+# autoware_ad_api_msgs/msg/SteeringFactor
+
+## Definition
+
+```txt
+# constants for common use
+uint16 UNKNOWN = 0
+
+# constants for type
+uint16 INTERSECTION = 1
+uint16 LANE_CHANGE = 2
+uint16 AVOIDANCE_PATH_CHANGE = 3
+uint16 AVOIDANCE_PATH_RETURN = 4
+uint16 STATION = 5
+uint16 PULL_OUT = 6
+uint16 PULL_OVER = 7
+uint16 EMERGENCY_OPERATION = 8
+
+# constants for direction
+uint16 LEFT = 1
+uint16 RIGHT = 2
+
+# constants for status
+uint16 APPROACHING = 1
+uint16 TRYING = 2
+uint16 TURNING = 3
+
+# variables
+geometry_msgs/Pose pose
+float32 distance
```

Comment:
should we make those into an array?
[Start pose, End Pose]
So that we can output the start pose and end pose when performing lane change and avoidance

Same as the distance, we output the distance from the start pose and also from end pose